packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -333,6 +334,19 @@ export default class Extension {
       pair.decodePkcs8(password);
     }
 
+    if (Object.prototype.hasOwnProperty.call(request.payload, 'genesisHash')) {
```

Comment:
The payload will always contain the genesisHash?

Reply:
There can be two different signing ways
**extrinsicSign** where payload is of type SignerPayloadJSON which contains the genesisHash
**bytesSign** (raw) where payload is of type SignerPayloadRaw which does not contain genesisHash (https://github.com/polkadot-js/api/blob/a9c9fb5769dec7ada8612d6068cf69de04aa15ed/packages/types/src/types/extrinsic.ts#L101)

Reply:
Indeed. I misread the next call (I blame GH context here) as a direct call to the `sign` method.

So in this case, instead of casting create a `isJsonPayload (value: SignerPayloadJSON | SignerPayloadRaw): valus is SignerPayloadJSON` function and use in in the if

Also suggest just doing `!isUndefined((request.payload as SignerPayloadJSON).genesisHash` as the check there.